nntrainer/include/layer.h

```diff
@@ -537,11 +537,16 @@ class Layer {
    * @param[in] type property type to be passed
    * @param[in] value value to be passed, if empty string is passed, do nothing
    * but throws error when @a type is invalid
+   * @exception std::out_of_range     when property type is not valid for the
+   * particular layer
```

Comment:
Is it out_of_range or invalid_parameter?

Reply:
it is out_of_range for this PR. Later gonna make custom exception for this to keep PR size small(#315)

Reply:
Patched at #320